### PR TITLE
Feat/merkle tree verify proof

### PR DESCRIPTION
**Pull request by LucasUTNFRD**
# Implement Merkle Tree Proof Verification

This PR implements the `verify_proof` function for the Merkle tree implementation, enabling verification of membership proofs.

## Changes

- Added `verify_proof` method to validate Merkle proofs
- Enhanced `MerkleProof` type to include directional information (left/right) for each proof hash
- Added comprehensive test cases for proof verification

### CODE REVIEW

File: src/merkle.rs [hunk 1]
```diff
@@ -13,7 +13,10 @@ struct MerkleTree {
     leaves: Vec<Hash>,    // This will be uses to add new elements to the tree and rebuild it
 }
 
-type MerkleProof = Vec<Hash>;
+/// Type alias for a Merkle proof
+/// A proof is a list of hashes that can be used to verify the membership of a leaf in the tree
+/// Each hash has associated a boolean value that indicates if the hash is a left or right sibling
+type MerkleProof = Vec<(Hash, bool)>;
```
Comment (dsocolobsky) [inline src/merkle.rs:19]: Then it should be an enum `Left`/`Right` I think, which adds more meaning to the value.

**Review by dsocolobsky — APPROVED** [review]
Approved but I left a comment which I think improves a bit the code.

File: src/merkle.rs (hunk 1 above)
Comment (lima-limon-inc) [reply src/merkle.rs:19]: I second this as well. Enums are our friend! 

Specially Rust enums, super useful feature.

**Review by lima-limon-inc — APPROVED** [review]
Great work

File: src/merkle.rs (hunk 1 above)
Comment (LucasUTNFRD) [reply src/merkle.rs:19]: Nice, enums will be used for the next PR.

**Pull request merged**